internal/pkg/gateway/api_test.go

```diff
@@ -77,515 +77,426 @@ var defaultEndpointDef = &endpointDef{
+			endpointDefinition: &endpointDef{
+				proposalError: status.Error(codes.Aborted, "mumbo-jumbo"),
 			},
-			{
-				name: "discovery fails",
-				plan: endorsementPlan{
-					"g1": {"localhost:7051"},
-				},
-				signedProposal: validSignedProposal,
-				setupDiscovery: func(d *mocks.Discovery) {
-					d.PeersForEndorsementReturns(nil, fmt.Errorf("mango-tango"))
-				},
-				errString: "mango-tango",
+			errString: "rpc error: code = Aborted desc = failed to evaluate transaction",
+			errDetails: []*pb.EndpointError{{
+				Address: "localhost:7051",
+				MspId:   "msp1",
+				Message: "rpc error: code = Aborted desc = mumbo-jumbo",
+			}},
+		},
+		{
+			name: "process proposal chaincode error",
+			plan: endorsementPlan{
+				"g1": {"peer1:8051"},
 			},
-			{
-				name: "process proposal fails",
-				plan: endorsementPlan{
-					"g1": {"localhost:7051"},
-				},
-				signedProposal: validSignedProposal,
-				endpointDefinition: &endpointDef{
-					proposalError: status.Error(codes.Aborted, "mumbo-jumbo"),
-				},
-				errString: "rpc error: code = Aborted desc = failed to evaluate transaction",
-				errDetails: []*pb.EndpointError{{
-					Address: "localhost:7051",
-					MspId:   "msp1",
-					Message: "rpc error: code = Aborted desc = mumbo-jumbo",
-				}},
+			endpointDefinition: &endpointDef{
+				proposalResponseStatus:  400,
+				proposalResponseMessage: "Mock chaincode error",
 			},
-			{
-				name: "process proposal chaincode error",
-				plan: endorsementPlan{
-					"g1": {"peer1:8051"},
-				},
-				endpointDefinition: &endpointDef{
-					proposalResponseStatus:  400,
-					proposalResponseMessage: "Mock chaincode error",
-				},
-				signedProposal: validSignedProposal,
-				errString:      "rpc error: code = Aborted desc = transaction evaluation error",
-				errDetails: []*pb.EndpointError{{
-					Address: "peer1:8051",
-					MspId:   "msp1",
-					Message: "error 400, Mock chaincode error",
-				}},
+			errString: "rpc error: code = Aborted desc = transaction evaluation error",
+			errDetails: []*pb.EndpointError{{
+				Address: "peer1:8051",
+				MspId:   "msp1",
+				Message: "error 400, Mock chaincode error",
+			}},
+		},
+		{
+			name: "dialing endorser endpoint fails",
+			plan: endorsementPlan{
+				"g1": {"peer2:9051"},
 			},
-			{
-				name: "dialing endorser endpoint fails",
-				plan: endorsementPlan{
-					"g1": {"peer2:9051"},
-				},
-				signedProposal: validSignedProposal,
-				setupRegistry: func(reg *registry) {
-					reg.endpointFactory.dialer = func(_ context.Context, target string, _ ...grpc.DialOption) (*grpc.ClientConn, error) {
-						if target == "peer2:9051" {
-							return nil, fmt.Errorf("endorser not answering")
-						}
-						return nil, nil
+			postSetup: func(def *preparedTest) {
+				def.dialer.Calls(func(_ context.Context, target string, _ ...grpc.DialOption) (*grpc.ClientConn, error) {
+					if target == "peer2:9051" {
+						return nil, fmt.Errorf("endorser not answering")
```

Comment:
nit: consider using error strings that are clearly coming from tests instead off production code. This reads like something I'd conceivably see in prod. The change could be as subtle as connecting the words with hyphens.